src/Altinn.Platform/Altinn.Platform.Storage/Storage/Helpers/InstanceHelper.cs

```diff
@@ -44,22 +45,17 @@ public static MessageBoxInstance ConvertToMessageBoxInstance(Instance instance)
                 ReadStatus = status.ReadStatus
             };
 
-            return messageBoxInstance;
-        }
-
-        /// <summary>
-        /// Adds title to the intance
-        /// </summary>
-        public static List<MessageBoxInstance> AddTitleToInstances(List<MessageBoxInstance> instances, Dictionary<string, Dictionary<string, string>> appTitles, string language)
-        {
-            foreach (MessageBoxInstance instance in instances)
+            if (instance.Status?.SubStatus != null)
             {
-                string title = appTitles[GetAppId(instance)].ContainsKey(language) ? appTitles[GetAppId(instance)][language] : appTitles[GetAppId(instance)]["nb"];
-                instance.Title = title;
+                messageBoxInstance.Substatus = new Substatus
+                {
+                    Label = instance.Status.SubStatus.Label,
+                    Description = instance.Status.SubStatus.Description
+                };
```

Comment:
Is this necessary? If so, then what about Title?

Reply:
Here we copy the text key for the substatus over to the message box instance, it is later replaced.
For the title we already know which text key it corresponds to.